### PR TITLE
hashtag-suggestion-in-postview

### PR DESCRIPTION
**Pull request by scoder1747**
## Summary

This PR enables a user to see  Hash Tag Suggestion List in the Post View while composing a note. When the user begins typing a word starting with '#', he will be able to see and scroll through the list of trending hash-tags (or at least default hash-tags if unavailable). This hash-tag data is collected when the user goes onto Search-page and stored it into a static property SuggestedHashtagsView.lastRefresh_hashtags. This approach is light-weight as no active network request is being triggered while the last refresh data is taken into account. After user taps one of the items in the list, the post will be appended with the clicked value (just as how @ tagging works currently).

Video demonstration:

https://github.com/user-attachments/assets/4829ce2b-13d8-435d-9361-eca1921d185a

_[Please provide a summary of the changes in this PR.]_

## Checklist

- [ ] I have read (or I am familiar with) the [Contribution Guidelines](../docs/CONTRIBUTING.md)
- [x] I have tested the changes in this PR
- [ ] My PR is either small, or I have split it into smaller logical commits that are easier to review
- [ ] I have added the signoff line to all my commits. See [Signing off your work](../docs/CONTRIBUTING.md#sign-your-work---the-developers-certificate-of-origin)
- [ ] I have added appropriate changelog entries for the changes in this PR. See [Adding changelog entries](../docs/CONTRIBUTING.md#add-changelog-changed-changelog-fixed-etc)
    - [ ] I do not need to add a changelog entry. Reason: _[Please provide a reason]_
- [ ] I have added appropriate `Closes:` or `Fixes:` tags in the commit messages wherever applicable, or made sure those are not needed. See [Submitting patches](https://github.com/damus-io/damus/blob/master/docs/CONTRIBUTING.md#submitting-patches)

## Test report

_Please provide a test report for the changes in this PR. You can use the template below, but feel free to modify it as needed._

**Device:** _[Please specify the device you used for testing]_

**iOS:** _[Please specify the iOS version you used for testing]_

**Damus:** _[Please specify the Damus version or commit hash you used for testing]_

**Setup:** _[Please provide a brief description of the setup you used for testing, if applicable]_

**Steps:** _[Please provide a list of steps you took to test the changes in this PR]_

**Results:**
- [ ] PASS
- [ ] Partial PASS
  - Details: _[Please provide details of the partial pass]_

## Other notes

_[Please provide any other information that you think is relevant to this PR.]_

### CODE REVIEW

**Comment by alltheseas** [comment]
@robagreda what do you think :eyes:

File: damus/Views/PostView.swift [hunk 1]
```diff
@@ -540,6 +550,23 @@ func get_searching_string(_ word: String?) -> String? {
...
+    }
+
+    guard word.count >= 2 else {
+        return nil
+    }
+    
+    guard let firstCharacter = word.first,
+          firstCharacter == "#" else {
+        return nil
+    }
```
Comment (jb55) [inline damus/Views/PostView.swift:565]:
```suggestion
    guard let word,
          word.count >= 2,
          let first_char = word.first,
          first_char == "#" else {
        return nil
    }
```

File: damus/Views/SuggestedHashtagsView.swift [hunk 2]
```diff
@@ -39,17 +39,63 @@ struct SuggestedHashtagsView: View {
...
+    @Binding var focusWordAttributes: (String?, NSRange?)
+    @Binding var newCursorIndex: Int?
+    @Binding var post: NSMutableAttributedString
+    @EnvironmentObject var tagModel: TagModel
+    
+    init(damus_state: DamusState,
+         suggested_hashtags: [String]? = nil,
+         max_items item_limit: Int? = nil,
+         events: EventHolder,
+         isFromPostView: Bool = false,
```
Comment (jb55) [inline damus/Views/SuggestedHashtagsView.swift:84]: this seems hacky, shouldn't this just be a different view entirely?

**Review by jb55 — CHANGES_REQUESTED** [review]

File: damus/Views/PostView.swift (hunk 1 above)
Comment (scoder1747) [reply damus/Views/PostView.swift]: I have made the update

**Pull request closed without merging**